File: Lab7_Python/Lab7/lab7.py
import math
func_1 = lambda n,x : (n**x)/math.factorial(x)

print("Please enter a value for n")
value_n=int(input())
print("Please enter a value for x")
value_x=int(input())

x=[]
total=0
for a in range(value_x):
    x.append(func_1(value_n,a))

for a in x:
    total+=a

print(total)
#end of question 1

total_func_2=0
def func_2 (temp_n):
    global total_func_2
    if (temp_n < 1):
        print(total_func_2)
    else:
        total_func_2+=(-1) ** (temp_n + 1) / temp_n
        func_2(temp_n-1)

func_2(5)

File: Lab7_C++/Lab_7/lab7.cpp
#include <iostream>
#include <cmath>
using namespace std;

double total_func_2=0;
double total_func_2_2=0;
bool started=true;
int temp_int_q4=0;

double func_2(int temp_n){
    if(temp_n < 1){
        return total_func_2;
    }
    else{
        total_func_2 += pow(-1, temp_n + 1) / temp_n;
        func_2(temp_n-1);
    }
}
// end of question 3

double func_2(){
    if(started){
        cout<<"Please enter a value "<<endl;
        cin>>temp_int_q4;
        started=false;
    }

    while(temp_int_q4>=1){
        total_func_2_2 += pow(-1, temp_int_q4 + 1) / temp_int_q4;
        temp_int_q4 -=1;
        func_2();
    }
    return total_func_2_2;

}
// end of question 4

int main() {
    cout<<func_2(7)<<endl;
    // end of question 3

    cout<<func_2();
    // end of question 4
}
